ansible_collections/arista/avd/roles/eos_designs/templates/connected_endpoints/monitor-sessions.j2

```diff
@@ -0,0 +1,39 @@
+{# Leaf connected_endpoint monitor-sessions #}
+monitor_sessions:
+{% for endpoint_key in connected_endpoints | arista.avd.natural_sort %}
```

Comment:
Instead of parsing the full data model just to look for monitor_session config, I suggest that we create another list like `connected_endpoints` - ex. `monitor_sessions_data` in the main.yml and then check for the monitor session stuff in the regular ethernet_interfaces and port_channel_interfaces templates and add the data to the list.
So ex. insert a new item into the list - remember to add an extra key for "interface". Ping me if this is unclear.

Reply:
@ClausHolbechArista Sorry, didn't have time to review this yet. I'll look into it after a short break and get back to you if needed. Thanks for the guidance already!

Reply:
Hopefully changed according to your suggestion. Please review if you find time. Will add the missing options to monitor-sessions.j2 within the next days